### ISSUE TITLE
Make `KRadioButton` to show a warning for developers when it's not nested inside `KRadioButtonGroup`

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri-design-system#761]
🌱 **Are you new to the codebase? Welcome! Please see [the contributing guidelines](https://github.com/learningequality/kolibri-design-system/blob/develop/CONTRIBUTING.md).**

## Summary

In https://github.com/learningequality/kolibri-design-system/pull/650, we introduced a new `KRadioButtonGroup` that ensures `KRadioButton` groups are accessible in all supported browsers. For this to properly work, all `KRadioButton`s need to be nested inside `KRadioButtonGroup` like

```
<KRadioButtonGroup>
  <KRadioButton>
  <KRadioButton>
</KRadioButtonGroup>
```

but it could also be more deeply nested structures such as

```
<KRadioButtonGroup>
  <KGridItem>
    <div>
      <KRadioButton>
      <KRadioButton>
    </div>
  </KGridItem>
</KRadioButtonGroup>
```

**The goal of this issue is to add logic to `KRadioButton` that shows a warning for developers when it's not nested inside `KRadioButtonGroup`.** 

## The Value Add

This will notify developers about the proper usage and ensures expected a11y behavior for users.

## Possible Tradeoffs

A risk of possible performance issues since it adds logic to every `KRadioButton` that needs to traverse all its parents (see "Guidance" on how to possibly take care of this)

## Guidance

- You can use [the playground page](https://github.com/learningequality/kolibri-design-system/blob/develop/dev_docs/03_how_to_update_library.md#1-preview-updates-in-the-kolibri-design-system-environment) for development and testing
- [`KRadioButton` documentation page](https://develop--kolibri-design-system.netlify.app/kradiobutton/) with more examples of usage
- As demonstrated in the examples above, note that we can't know how deeply `KRadioButton` is nested in `KRadioButtonGroup` so we will need to look at its chain of parents until `KRadioButtonGroup` is found.
  - However, as this will be run from every `KRadioButton`, we need to be careful to not use techniques of accessing the DOM that may cause performance issues.
  - Recursion over parents in `this.$parent` and checking each `$parent`'s `$options._componentTag` until we find `'KRadioButtonGroup'` component tag or until there is no more parents, could be the way
- Lastly, there's no need to run the check in production so let's wrap it in `if (process.env.NODE_ENV !== 'production') {`. @rtibbles says "this will also tree shake the code out in webpack builds." ;)

## Acceptance criteria

- [ ] No regressions in `KRadioButton`s
- [ ] Optimized for performance
- [ ] Executes in the development mode only
- [ ] Works for one level of nesting as well as multiple levels and has unit tests for these two scenarios

## Comments

This update would ideally be released after https://github.com/learningequality/kolibri/issues/12596 so that we're not overwhelmed by warnings in Kolibri ;)

**Comment by lokesh-sagi125** [comment]
hey @MisRob  can i work on this?

**Comment by lokesh-sagi125** [comment]
hey @MisRob can i work on this issue?

**Comment by MisRob** [comment]
Hi @lokesh-sagi125, you're welcome to take this on. Thanks for your continued contributions. Please follow guidance closely and you're welcome to ask questions here if there's anything that's not clear.

**Comment by lokesh-sagi125** [comment]
hey @MisRob you want the warning to be displayed in the console right?

**Comment by MisRob** [comment]
Hi @lokesh-sagi125, yes

**Comment by iamshobhraj** [comment]
@MisRob can i work on this?

**Comment by AlexVelezLl** [comment]
Hey @iamshobhraj! Thanks for your interest in contributing to this issue. Unfortunately, this is already assigned. But you’re welcome to find [a "help wanted" issue with no assignee](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues) :hugs:.

**Comment by lokesh-sagi125** [comment]
hey @MisRob i am getting a warning saying 'event' is deprecated , should i use 'Event' instead or since this issue has no connection with event should just leave it as is?
<img width="1020" alt="Screenshot 2024-09-15 at 12 33 02 PM" src="https://github.com/user-attachments/assets/3c493f8d-e074-4860-b066-c9458e6583b6">

**Comment by lokesh-sagi125** [comment]
hey @MisRob sorry for the inactivity i was busy with a hackathon ; i just opened a PR for this issue could you please review the changes :).

**Comment by AlexVelezLl** [comment]
Closed by #781
